File: java/Gate Three/TestQuagmire.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class TestQuagmire {

    @Test
    public void testSpecifyListLengthToAGreaterNumbers() {
	int [] array = {1, 2, 3};
	int[] actual = Quagmire.specifyListLength(array, 5);
	int[] expected = {1, 2, 3, -1, -1};
	assertArrayEquals(actual, expected);
	}
	
   @Test
    public void testSpecifyListLengthToALesserNumbers() {
	int [] array = {1, 2, 3, 4, 5, 6};
	int[] actual = Quagmire.specifyListLength(array, 3);
	int[] expected = {1, 2, 3};
	assertArrayEquals(actual, expected);
	}

   @Test
    public void testSpecifyListLengthEdgeCase() {
	int [] array = {};
	int[] actual = Quagmire.specifyListLength(array, 7);
	int[] expected = {-1, -1, -1, -1, -1, -1, -1};
	assertArrayEquals(actual, expected);
	}
}

File: python/Gate Three/TestQuagmire.py
import unittest
from quagmire import *

class TestQuagmire(unittest.TestCase):
    def test_specify_list_length_to_a_greater_numbers(self):
        actual = specify_list_length([1, 2, 3], 5)
        expected = [1, 2, 3, -1, -1]
        self.assertEqual(actual, expected)
    def test_specify_list_length_to_a_lesser_numbers(self):
        actual = specify_list_length([1, 2, 3, 4, 5, 6], 3)
        expected = [1, 2, 3]
        self.assertEqual(actual, expected)
    def test_specify_list_length_edge_case(self):
        actual = specify_list_length ([], 7)
        expected = [-1, -1, -1, -1, -1, -1, -1]
        self.assertEqual(actual, expected)
